src/NUnitEngine/nunit.engine.tests/Runners/MasterTestRunnerTests.cs

```diff
@@ -21,53 +21,130 @@
+            new AssemblyData("mock-assembly.dll", "Runnable", MockAssembly.Tests, MockAssembly.PassedInAttribute, MockAssembly.Failed, MockAssembly.Skipped, MockAssembly.Inconclusive);
+
+        static AssemblyData NoTestsAssemblyData =
+            new AssemblyData("notests-assembly.dll", "NotRunnable", 0, 0, 0, 0, 0);
+
+        static AssemblyData UnknownProjectType =
+            new AssemblyData("project.xxx", "NotRunnable", 0, 0, 0, 0, 0);
+
+        static AssemblyData EmptyProject =
+            new AssemblyData("empty.nunit", "NotRunnable", 0, 0, 0, 0, 0);
+
+        static AssemblyData ProjectWithMockAssembly =
+            new AssemblyData("project.nunit", "Runnable", MockAssembly.Tests, MockAssembly.PassedInAttribute, MockAssembly.Failed, MockAssembly.Skipped, MockAssembly.Inconclusive);
+
+        static TestRunData[] FixtureData = new TestRunData[]
+        {
+            // NOTES: 
+            // 1. Currently, these tests document actual behavioral differrences
+            // among the different builds. We may want to change that.
+            // 2. The empty tests are excluded for .NET Core because of 
+            // errors and different behavior which either needs to be accounted 
+            // for in the tests or corrected.
+            // 3. .NET Standard 1.6 doesn't support projects.
+#if NETCOREAPP1_1
+            new TestRunData( typeof(LocalTestRunner), MockAssemblyData ),
+            new TestRunData( typeof(AggregatingTestRunner), MockAssemblyData, MockAssemblyData ),
+            new TestRunData( typeof(LocalTestRunner), UnknownProjectType ),
+#elif NETCOREAPP2_0
+            new TestRunData( typeof(LocalTestRunner), MockAssemblyData ),
+            new TestRunData( typeof(AggregatingTestRunner), MockAssemblyData, MockAssemblyData ),
+            new TestRunData( typeof(LocalTestRunner), UnknownProjectType ),
+            new TestRunData( typeof(AggregatingTestRunner), EmptyProject ),
+            new TestRunData( typeof(AggregatingTestRunner), ProjectWithMockAssembly ),
+#else
+            new TestRunData( typeof(TestDomainRunner), MockAssemblyData ),
+            new TestRunData( typeof(MultipleTestDomainRunner), MockAssemblyData, MockAssemblyData ),
+            new TestRunData( typeof(TestDomainRunner), UnknownProjectType ),
+            new TestRunData( typeof(TestDomainRunner), EmptyProject ),
+            new TestRunData( typeof(TestDomainRunner), ProjectWithMockAssembly ),
+            new TestRunData( typeof(TestDomainRunner), NoTestsAssemblyData ),
+            new TestRunData( typeof(MultipleTestDomainRunner), NoTestsAssemblyData, NoTestsAssemblyData ),
+            new TestRunData( typeof(MultipleTestDomainRunner), MockAssemblyData, NoTestsAssemblyData )
+#endif
+        };
+
         [SetUp]
         public void Initialize()
         {
             _package = new TestPackage(_assemblies);
+            _package.AddSetting(EnginePackageSettings.ProcessModel, "InProcess");
```

Comment:
This was using `InProcessTestRunnerFactory`, as many of our tests do, in order to avoid creating a process. I realized that using `InProcessTestRunnerFactory` directly bypasses an important bit of code in `DefaultTestRunnerFactory`, which is where we recognize whether there are any projects listed in the command-line. The result is that the tests actually behave differently from how the console app behaves.

In this case, I replaced `InProcessTestRunnerFactory` with `DefaultTestRunnerFactory` and set the `ProcessModel` to `InProcess`. This gives the correct behavior.

I think we should review and fix any other tests that use this technique but also review the way we use InProcessTestRunner in our agent process. We may need to refactor so that the examination of how many projects we have occurs either in both runners or separately.

Reply:
Sounds good. Could you log an issue to look at the tests?